### PR TITLE
Move roomList out of MatrixClient, into legacy Crypto

### PR DESCRIPTION
**Pull request by andybalaam**
Part of https://github.com/vector-im/element-web/issues/26108

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Review by dbkr — APPROVED** [review]

File: src/crypto/index.ts [hunk 1]
```diff
@@ -473,10 +474,14 @@ export class Crypto extends TypedEventEmitter<CryptoEvent, CryptoEventHandlerMap
...
         private readonly clientStore: IStore,
         public readonly cryptoStore: CryptoStore,
-        private readonly roomList: RoomList,
         verificationMethods: Array<VerificationMethod | (typeof VerificationBase & { NAME: string })>,
     ) {
         super();
+
+        logger.debug("Crypto: initialising roomlist...");
+        this.roomList = new RoomList(cryptoStore);
+        this.roomList.init();
```
[the remark this answers is not in the document]
Comment (andybalaam) [reply src/crypto/index.ts]: Thanks, fixed in 738113375

**Comment by andybalaam** [comment]
@dbkr the SonarCloud errors are about 2 code smells stating that I am using deprecated APIs. Since this change is 100% in deprecated code, I think that is allowable, so please could you force-merge if you agree?

**Comment by andybalaam** [comment]
> Why force merge... Just dismiss the alerts as won't fix or false positive.

I don't have that button:

image

**Comment by t3chguy** [comment]
@andybalaam have you tried logging in? The button would not be available to those not logged in.

**Comment by andybalaam** [comment]
> @andybalaam have you tried logging in? The button would not be available to those not logged in.

Thanks, that fixed it.

**Pull request merged**